h/views/api_auth.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from datetime import (datetime, timedelta)
+
+import jwt
+
+from h import models
+from h.util.view import json_view
+
+
+class OAuthTokenError(Exception):
+    def __init__(self, message, description=None, status_code=400):
+        self.description = description
+        self.status_code = status_code
+        super(OAuthTokenError, self).__init__(message)
+
+
+@json_view(route_name='token', request_method='POST')
+def access_token(request):
+    grant_token = _validate_request(request)
+    userid = _extract_userid(request, grant_token)
+
+    user = request.db.query(models.User).filter_by(userid=userid).one_or_none()
```

Comment:
It's probably a good idea to use `UserService.fetch` to retrieve users by userid.